pennylane/transforms/split_qscript.py

```diff
@@ -311,7 +192,17 @@ def sum_expand(qscript: QuantumScript, group=True):
                     idxs_coeffs_dict[s_m.hash].append((idx, coeff))
             continue
 
-        coeff = 1 if isinstance(m, ExpectationMP) else None
+        if isinstance(obs, Hamiltonian) and isinstance(m, ExpectationMP):
+            for o, coeff in zip(obs.ops, obs.data):
+                o_m = qml.expval(o)
+                if o_m.hash not in measurements_dict:
+                    measurements_dict[o_m.hash] = o_m
+                    idxs_coeffs_dict[o_m.hash] = [(idx, coeff)]
+                else:
+                    idxs_coeffs_dict[o_m.hash].append((idx, coeff))
+            continue
```

Comment:
Is it the right indentation for continue?

Reply:
Changed this `continue` statements with `if` and `elif` statements.

I believe now it is clearer